diesel_migrations/migrations_internals/src/migration.rs

```diff
@@ -54,9 +54,15 @@ impl<'a> fmt::Display for MigrationFileName<'a> {
     }
 }
 
-pub fn migration_from(path: PathBuf) -> Result<Box<dyn Migration>, MigrationError> {
+pub fn migration_from<Conn>(
+    conn: &Conn,
+    path: PathBuf,
+) -> Result<Box<dyn Migration>, MigrationError>
+where
+    Conn: MigrationConnection,
+{
     #[cfg(feature = "barrel")]
-    match ::barrel::integrations::diesel::migration_from(&path) {
+    match ::barrel::integrations::diesel::migration_from(conn, &path) {
```

Comment:
The current `barrel` doesn't support this `conn` param. I can't update the version of barrel because barrel's version depends on this change being published.